### PR TITLE
CleanESD for MC reconstruction

### PR DESCRIPTION
**Pull request by catalinristea**
Ciao @chiarazampolli, @fprino, @shahor02 

Here is a CleanESD proposal for MC reconstruction, as discussed in ALIROOT-8418. Please have a look and let me know if I missed anything. Thanks!

Yours,
Catalin.

### CODE REVIEW

File: MC/ReconstructionConfig.C [hunk 1]
```diff
@@ -269,10 +269,44 @@ void ReconstructionDefault(AliReconstruction &rec)
...
+        if(system.EqualTo("p-p"))
+            if(passname == "pass1" ||
+               (passname == "pass2" && (periodname == "LHC16k" || periodname == "LHC16l")) ||
+               ((passname == "pass1" || passname == "pass2" || passname == "pass3" || passname == "pass4") && periodname == "LHC15n")
+               )
+                cleanESD = kFALSE;
+    
+        if(system.EqualTo("p-Pb") || system.EqualTo("Pb-p"))
+            if(passname == "pass1" ||
+               (year == 2013 && (passname == "pass1" || passname == "pass2") || passname == "pass3" || passname == "pass4" || passname == "pass5")
```
Comment (chiarazampolli) [inline MC/ReconstructionConfig.C:295]: Shouldn't this:

` (year == 2013 && (passname == "pass1" || passname == "pass2") || passname == "pass3" || passname == "pass4" || passname == "pass5")`

become:
`(year == 2013 && (passname == "pass1" || passname == "pass2" || passname == "pass3" || passname == "pass4" || passname == "pass5"))`

I changed the parenthesis, to include all possible passes in the same pair.

**Comment by shahor02** [comment]
Hi @catalinristea 

The logins looks ok to me, but then that are some periods done obviously w/o cleanESD which are not in your disabling list, e.g LHC15X_pass2 protons except LHC15n_pass2. Is it because we know that they will be never asked to be simulated?

Cheers,
 Ruben

**Comment by catalinristea** [comment]
Hi Ruben,

Thanks, I would then add 15h_pass2 and 15i_pass2 to this list (as 15l_pass1 is already covered), as reco was also done w/o CleanEsd.

Yours,
Catalin.

**Comment by catalinristea** [comment]
Ciao Chiara,

Regarding the data, in main_rec.C we have ALIEN_JDL_LPMESDCLEANOFF which is used to turn OFF ESD cleaning. By default we don't provide it via JDL, hence the default is ON.
Thanks!

Yours,
Catalin.

**Comment by chiarazampolli** [comment]
Ciao @catalinristea , 

Perfect then! For the next reprocessings, it will be on.

Chiara

File: MC/ReconstructionConfig.C (hunk 1 above)
Comment (catalinristea) [reply MC/ReconstructionConfig.C:295]: Ciao Chiara,

Sure, this was a typo, thanks for spotting it!

Yours,
Catalin.

**Comment by chiarazampolli** [comment]
Ciao,

For me it is fine to merge. @fprino , @shahor02 , do you have further comments?

Chiara

**Comment by shahor02** [comment]
fine for me.

**Comment by fprino** [comment]
Hi,

fine also with me. Thanks Catalin for taking care.

Ciao, Francesco

**Comment by chiarazampolli** [comment]
Merged, thanks again @catalinristea !

**Pull request merged**